tests/fixture/test_data_server.go

```diff
@@ -37,15 +38,14 @@ type TestData struct {
 
 // http server allocating resources to tests potentially executing in parallel
 type testDataServer struct {
-	TestData
-
 	// Synchronizes access to test data
 	lock sync.Mutex
+	TestData
```

Comment:
I think it's a general best practice to put the `lock` over the fields that it is locking. I don't recall where I read this... But I think it makes sense... It isn't as important here... But in cases where the struct gets larger then it can improve readability.